translator.py

```diff
@@ -0,0 +1,110 @@
+"""Translates any text document into an .fb file
+
+First CLI argument should be a text file to be converted.
+Print to `.fb` file extension with ` > output_file.fb`
+
+Example:
+
+./translator.py original_text_file.txt > new_fb_file.fb
+
+Check translation with ./code_runner.py new_fb_file.fb"""
+
+import sys
+
+
+class Translator():
+    """Prepares common FizzBuzzLang functions and retrieves file text content
+
+    ---
+    Attributes:
+        look_up: dict -- self-populating dictionary of finary refs
+        INPUT: str -- text content of file passed in CLI
+        STORE_CHAR: str -- allows storing of character at data-space location
+        MOVE_FORWARD: str -- move pointer forward one space in data-space
+        STORE_POS_FIZZ: str -- stores current data-space location in FIZZ loc.
+        END_OF_CODE: str -- prints data
+        TERMINATE_COMMAND: str -- termination command to end fbi
+        FBL_TEXT_ARRAY: list -- INPUT written in finary
+    """
+
+    def __init__(self) -> None:
+        self.look_up = {}
+        self.INPUT = self._retrieve_file()
+        self.STORE_CHAR = "BUZZ FIZZBUZZ FIZZBUZZ "
+        self.MOVE_FORWARD = "FIZZ FIZZ FIZZ"
+        self.STORE_POS_FIZZ = "FIZZ FIZZBUZZ FIZZ"
+        self.END_OF_CODE = """BUZZ FIZZBUZZ FIZZBUZZ BUZZ FIZZ BUZZ FIZZ
+FIZZ FIZZBUZZ FIZZBUZZ FIZZ
+FIZZBUZZ FIZZ FIZZBUZZBUZZ
+BUZZ BUZZ
+FIZZ FIZZ FIZZ
+FIZZBUZZ BUZZ FIZZ FIZZBUZZBUZZ"""
+        self.TERMINATE_COMMAND = "FIZZBUZZ FIZZBUZZ"
+        self.FBL_TEXT_ARRAY = self._convert_to_fbl()
+
+    def _retrieve_file(self):
+        """Retrieves text from file
+
+        Accepts CLI argument
+
+        ---
+        returns:
+            str: -- document contents as string
+
+        raises:
+            IndexError: -- if no file name provided
+            FileNotFoundError: -- if file is not found
+
+            All errors will terminate program
+        """
+
+        try:
+            file_name = sys.argv[1]
+        except IndexError:
+            print("Error, no file argument given when running file.")
+            print("Program will terminate.")
+            exit()
+
+        while True:
+            try:
+                with open(file_name, "r") as f:
+                    INPUT = f.read()
+                    break
+            except FileNotFoundError as e:
+                print(f"{e}")
+                print("Program will terminate.")
+                exit()
+        return INPUT
+
+    def _convert_to_fbl(self):
+        """Creates a list of FBL characters
+
+        ---
+        returns:
+            list: -- list of FBL converted ASCII characters
+        """
+
+        fbl_text = []
+        for char in self.INPUT:
+            char_finary = self.look_up.setdefault(char, " ".join(
+                ["FIZZ" if char == "0" else "BUZZ" for char in bin(
+                    ord(char))[2:]]))
+            fbl_text.append(self.STORE_CHAR + char_finary)
+
+        return fbl_text
+
+    def print_fbl_doc(self):
```

Comment:
I know it was actually me who originally suggested the prints, but looking at it now, I would suggest you return a string here, which the caller can do with as they please.
More specifically, it might be elegant to make this the `__str__` method. And if you want to keep the current nice code structure, you could replace the prints with writes to `io.StringIO`

Reply:
At present, I am easily able to write the output to a new `.fb` file. I think I will need to research a little more regarding the methods you suggest. I suppose a user may want to input this directly into another program without the need for the interim file, but would still be able to choose to write it directly to a new document if they desired.

Reply:
Exactly.
The general heuristic I'm following here is of Separation of Concerns, and specifically, a single function shouldn't deal with both business logic and I/O.